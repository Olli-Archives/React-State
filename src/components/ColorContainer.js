import React, { PureComponent } from 'react';
import ColorDiv from './ColorDiv';
import ColorPicker from './ColorPicker';

export default class ColorContainer extends PureComponent{
  state={
    color:''
  }
 
  onClickHandler = (newColor)=>{
    this.setState({ color: newColor });
    console.log(this.state.color);
  }



  render(){
    return (
      <div>
        <ColorPicker color='red' onClick={this.onClickHandler}/>
        <ColorPicker color='blue' onClick={this.onClickHandler}/>
        <ColorPicker color='green' onClick={this.onClickHandler}/> 
        <ColorDiv color={this.state.color}/>
      </div>    
    );
  }
}
