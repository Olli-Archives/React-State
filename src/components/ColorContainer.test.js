import React from 'react';
import { shallow } from 'enzyme';
import ColorContainer from './ColorContainer';

describe('color conainer component', ()=>{
  it('renders color container', ()=>{
    const wrapper = shallow(<ColorContainer/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
