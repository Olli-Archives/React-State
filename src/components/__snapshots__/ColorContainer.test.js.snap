// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color conainer component renders color container 1`] = `
<div>
  <ColorPicker
    color="red"
    onClick={[Function]}
  />
  <ColorPicker
    color="blue"
    onClick={[Function]}
  />
  <ColorPicker
    color="green"
    onClick={[Function]}
  />
  <ColorDiv
    color=""
  />
</div>
`;
